tests/suite/graceful_recovery_test.go

```diff
@@ -0,0 +1,206 @@
+package suite
+
+import (
+	"fmt"
+	"net/http"
+	"os/exec"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("Graceful Recovery test", Ordered, Label("nfr", "graceful-recovery"), func() {
+	files := []string{
+		"graceful-recovery/cafe.yaml",
+		"graceful-recovery/cafe-secret.yaml",
+		"graceful-recovery/gateway.yaml",
+		"graceful-recovery/cafe-routes.yaml",
+	}
+	ns := &core.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "graceful-recovery",
+		},
+	}
+
+	var teaURL, coffeeURL string
+
+	BeforeAll(func() {
+		teaURL = "https://cafe.example.com/tea"
+		coffeeURL = "http://cafe.example.com/coffee"
+		if portFwdPort != 0 {
+			teaURL = fmt.Sprintf("https://cafe.example.com:%s/tea", strconv.Itoa(portFwdPort))
```

Comment:
that's code I used from the `sample_test.go`. However, decided to remove this part as since this is an NFR test and should only be run on GKE, it should not ever be port forwarded so this check is useless.